### ISSUE TITLE
How to get a new instance of the Settings struct?

### ISSUE THREAD

**Issue by kinire98** [de-vri-es/serial2-rs#9]
I'm trying to open a port with Settings but there isn't seem to be any static method of the struch which returns the struct itself. Am I missing something or is there a way to initialize it?

**Comment by kinire98** [comment]
I can make a simple PR with the changes...

**Comment by kinire98** [comment]
Ok, nvm. This happens for not reading the docs enough. jeje. The answer is to open the port with:
```rust
let port = SerialPort::open("COM1", 9600);
``` 
and then get the configuration with: 
```rust
let settings = port.get_configuration();
```
Leaving this here if it helps someone. Sorry for bothering

**Comment by de-vri-es** [comment]
Yeah, to clarify why there is no `Settings::new()`: you should always preserve settings you don't care/know about. So instead of mmaking a "clean" settings struct and applying it, you always have to get one from the device, modify it, and set it again.

**Comment by brainstorm** [comment]
I also found this builder very confusing... in fact I'm still a bit confused on how to i.e set the flow control back to the `port` when the result type is unit `()`?:

![Screenshot 2023-08-03 at 9 34 41 pm](https://github.com/de-vri-es/serial2-rs/assets/175587/6b706ed1-c944-423e-9978-ae99bacb2d4d)

Can you provide some illustrative examples on how you are using it, @de-vri-es? Right now I'm observing on my oscilloscope that some packets are not being sent out to my (cheap, most probably counterfeit) FTDI clone and I'd like to pin down what is it through tweaking settings (and in the future set/unset RTS at will, if possible).... but I find this non-new() builder pattern a bit unfamiliar TBH :_S

**Comment by de-vri-es** [comment]
`settings.set_flow_control(...)` simply modifies the existing settings object.

```rust
let mut port = SerialPort::open("/dev/ttyUSB0", 115200)?;
let mut settings = port.get_configuration();
settings.set_flow_control(serial2::FlowControl::RtsCts)?;
port.set_configuration(&settings)?;
```

**Comment by brainstorm** [comment]
Thanks! I changed it as advised (don't know why I got stuck on that, doh):

```rust
	dbg!("Serial port setup...");
	// Serial port
	let mut port = SerialPort::open(PORT, BAUD)?;
	let mut settings = port.get_configuration()?;

	settings.set_flow_control(serial2::FlowControl::RtsCts);
	port.set_configuration(&settings)?;
```

But now I'm getting the following error:

```
[src/main.rs:53] "Serial port setup..." = "Serial port setup..."
Error: Custom { kind: Other, error: "failed to apply some or all settings" }
```

I will now check the `serial2` code and see what's the underlying issue, perhaps the dongle's (OSX) driver not allowing that?. That setting/change works fine in [CoolTerm](https://freeware.the-meiers.org/CoolTermMac.dmg).

**Comment by brainstorm** [comment]
Interesting:

```rust
	dbg!("Serial port setup...");
	// Serial port
	let mut port = SerialPort::open(PORT, BAUD)?;
	let mut settings = port.get_configuration()?;
	dbg!(&settings);
	settings.set_flow_control(serial2::FlowControl::RtsCts);
	dbg!(&settings);
	port.set_configuration(&settings)?;
```

Yields the following, which seem correctly applied settings to me 🤔 :

```
[src/main.rs:53] "Serial port setup..." = "Serial port setup..."
[src/main.rs:57] &settings = Settings {
    baud_rate: Ok(
        9600,
    ),
    char_size: Ok(
        Bits8,
    ),
    stop_bits: Ok(
        One,
    ),
    parity: Ok(
        None,
    ),
    flow_control: Ok(
        None,
    ),
}
[src/main.rs:59] &settings = Settings {
    baud_rate: Ok(
        9600,
    ),
    char_size: Ok(
        Bits8,
    ),
    stop_bits: Ok(
        One,
    ),
    parity: Ok(
        None,
    ),
    flow_control: Ok(
        XonXoff,
    ),
}
Error: Custom { kind: Other, error: "failed to apply some or all settings" }
```

**Comment by de-vri-es** [comment]
Yeah, not all devices support flow control. It's probably a limitation of the device.

**Comment by brainstorm** [comment]
Ok, interesting, I'd expect the altered settings to be `None` on that field instead of another (opposite?) value :-S

Is there a way to list the capabilities of a device? Where are those settings really stored?

**Comment by de-vri-es** [comment]
> Yields the following, which applies the **other** available flow control setting instead of the one I specified in my code thinking :

Ah, this is certainly a bug, although I thought I fixed that recently.

I'm not aware of any way to list device capabilities. But in order to see if the settings are accepted by the device you have to call `set_configuration()`, and if you want to be sure what is set afterwards you have to call `get_configuration()` again.

**Comment by de-vri-es** [comment]
> Ah, this is certainly a bug, although I thought I fixed that recently.

Released version 0.2.2 with a fix for this. Thanks for reporting it!
